cmd/job-companion/main.go

```diff
@@ -80,157 +81,189 @@ func main() {
+				if err := logJobSteps(c, jobID); err != nil {
 					log.Printf("Can't print steps info err: %s", err)
 				}
 
 				return errors.New("job canceled")
-			case slurm.JobStatusCompleted:
+			case api.JobStatus_COMPLETED:
 				if cOps != nil {
-					return collectResults(c, id, cOps)
+					return collectResults(c, cOps)
 				}
 				return nil
 			}
 		}
 	}
 }
 
-func logErrOutput(c slurm.Slurm, path string) error {
-	f, err := c.Open(path)
-	if err != nil {
-		return err
-	}
-
-	logs, err := ioutil.ReadAll(f)
+func logErrOutput(c api.WorkloadManagerClient, path string) error {
+	f, err := c.OpenFile(context.Background(), &api.OpenFileRequest{Path: path})
 	if err != nil {
 		return err
 	}
+	defer f.CloseSend()
 
 	log.Printf("Stderr output from %s", path)
-	log.Println(string(logs))
-	return nil
+	for {
+		chunk, err := f.Recv()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+
+			return errors.Wrap(err, "can't receive chunk")
+		}
+
+		log.Print(string(chunk.Content))
+	}
 }
 
-func logJobSteps(c slurm.Slurm, id int64) error {
-	steps, err := c.SJobSteps(id)
+func logJobSteps(c api.WorkloadManagerClient, id int64) error {
+	stepsResp, err := c.JobSteps(context.Background(), &api.JobStepsRequest{JobId: id})
 	if err != nil {
 		return err
 	}
 
-	for _, i := range steps {
-		log.Printf("JobID:%s State:%s ExitCode:%d Name:%s", i.ID, i.State, i.ExitCode, i.Name)
+	for _, i := range stepsResp.JobSteps {
+		log.Printf("JobID:%s State:%s ExitCode:%d Name:%s",
+			i.Id,
+			api.JobStatus_name[int32(i.Status)],
+			i.ExitCode,
+			i.Name,
+		)
 	}
 	return nil
 }
 
-func tailLogs(ctx context.Context, c slurm.Slurm, logFile string) chan struct{} {
+func tailLogs(ctx context.Context, c api.WorkloadManagerClient, logFile string) chan struct{} {
 	done := make(chan struct{})
 
 	go func() {
 		defer close(done)
-
-		f, err := c.Tail(logFile)
+		tf, err := c.TailFile(context.Background())
 		if err != nil {
-			log.Printf("Can't tail file %s err: %s", logFile, err)
+			if err != io.EOF {
+				log.Printf("Can't tail file %s err: %s", logFile, err)
+			}
 			return
 		}
-		defer f.Close()
+		defer tf.CloseSend()
+		if err := tf.Send(&api.TailFileRequest{Path: logFile, Action: api.TailAction_Start}); err != nil {
+			log.Printf("Can't send tail request err: %s", err)
+		}
 
 		buffCh := make(chan []byte)
 
-		// since reading from f is blocking we need to do it in a separate gorutine
+		// since reading from tf is blocking we need to do it in a separate gorutine
 		go func() {
+			defer close(buffCh)
 			for {
-				buff := make([]byte, 128)
-				n, err := f.Read(buff)
+				chunk, err := tf.Recv()
 				if err != nil {
 					return
 				}
 
-				buffCh <- buff[:n]
+				buffCh <- chunk.Content
 			}
 		}()
 
+		var waitingEOF bool
 		for {
 			select {
 			case <-ctx.Done():
-				log.Println("Tail logs finished from context")
-				return
-			case chunk := <-buffCh:
-				os.Stdout.Write(chunk)
+				if waitingEOF {
```

Comment:
I think it is better to disable this case by setting done channel to nil instead of using such flags. E.g.
```
done := ctx.Done()

case <-done:
    done  = nil
    // logic here
```